assembly-syntax/src/parser/error.rs

```diff
@@ -283,6 +283,21 @@ pub enum ParsingError {
         #[label]
         span: SourceSpan,
     },
+    #[error("invalid slice constant")]
+    #[diagnostic()]
+    InvalidSliceConstant {
+        #[label("slicing could be applied only to the word constants")]
+        #[label]
+        span: SourceSpan,
+    },
+    #[error("invalid word slice range")]
+    #[diagnostic()]
+    InvalidWordSliceRange {
```

Comment:
I think I'd prefer to split this into two variants:

* `EmptySlice` - raised when the slice range is empty, and therefore would produce no value.
* `InvalidRange` - raised when the range bounds are either inverted (i.e. start follows end) or one of the bounds is invalid (i.e. overflows `u8::MAX`)